src/components/PipelineRunDetailsView/visualization/utils/pipelinerun-graph-utils.ts

```diff
@@ -118,6 +118,29 @@ export const appendStatus = (
     } else if (mTask.status && !mTask.status.reason) {
       mTask.status.reason = RunStatus.Pending;
     }
+
+    // Determine any task test status
+    const taskRunName =
+      pipelineRun.status.taskRuns &&
+      Object.keys(pipelineRun.status.taskRuns).find(
+        (key) => pipelineRun.status.taskRuns[key].pipelineTaskName === mTask.name,
+      );
+    const taskRun = taskRunName && pipelineRun.status.taskRuns[taskRunName];
+    if (taskRun?.status?.taskResults) {
+      const testOutput = taskRun?.status?.taskResults.find(
+        (result) => result.name === 'HACBS_TEST_OUTPUT',
+      );
+      if (testOutput) {
+        try {
+          const outputValues = JSON.parse(testOutput.value);
+          if (outputValues.result === 'FAILURE' || outputValues.result === 'WARNING') {
+            mTask.testStatus = RunStatus.Cancelled;
```

Comment:
Does it mean that the warnings and failures are visually identical in the UI?

It would be nice if the user could easily see the difference between a warning and a failure, since the outcome is likely to be quite different. E.g. a build could be allowed to be deployed if there's a warning, but would be blocked if there is a failure.

Reply:
Just noticed the other comments above about this topic.

I still do think user should be able to see visually the difference between a warning and failure in HACBS_TEST_OUTPUT, but having the failed test result look similar to a failed taskrun could lead to confusion.

Could there somehow be two levels of warning icon, e.g. "minor warning" for warnings in HACBS_TEST_OUTPUT, "major warning" for failures?

Reply:
Waiting on UX to come up with a way to differentiate these. @andrew-ronaldson